_posts/2015-05-22-authentication-end-point.markdown

```diff
@@ -0,0 +1,103 @@
+
+Examples of integrations possible:
+
+* OAuth Login - [GitHub](https://github.com/srinivasupadhya/gocd-oauth-login), [Google](https://github.com/srinivasupadhya/gocd-oauth-login), Hotmail, Yahoo! etc.
+* Single Sign-on (SSO) - LDAP, Okta etc.
+* 2-factor authentication - SMS verification etc.
+* Custom `username` & `password` authentication
+
+### How does it work?
+
+Below is an explanation of how [GitHub OAuth Login plugin](https://github.com/srinivasupadhya/gocd-oauth-login) works.
+
+* Generate OAuth token on GitHub.
+
+<figure>
+  <img src="/images/blog/authentication-plugins/generate-oauth-token.png" class="has_border full_size"
+    alt="Figure 1: GitHub - Generate Oauth Token" id="mature_ci_cd_setup" title="GoCD - Generate Oauth Token" />
+  <figcaption>Figure 1: Generate oauth token <span class="click_to_enlarge">(Click to enlarge)</span></figcaption>
+</figure>
+
+* On plugin listing page users will see a gear icon (similar to one on the pipeline dashboard).
+
+<figure>
+  <img src="/images/blog/authentication-plugins/list-plugin.png" class="has_border full_size"
+    alt="Figure 1: GoCD - Plugin Listing" id="mature_ci_cd_setup" title="GoCD - Plugin Listing" />
+  <figcaption>Figure 1: Plugin listing with gear icon <span class="click_to_enlarge">(Click to enlarge)</span></figcaption>
+</figure>
+
+* Clicking on the gear icon opens a pop-up that renders "Plugin Settings".
+
+<figure>
+  <img src="/images/blog/authentication-plugins/configure-plugin.png" class="has_border full_size"
+    alt="Figure 2: GoCD - Configure Plugin" id="mature_ci_cd_setup" title="GoCD - Configure Plugin" />
+  <figcaption>Figure 2: Configure plugin pop-up <span class="click_to_enlarge">(Click to enlarge)</span></figcaption>
+</figure>
+
+* Access `<go-server>/go/plugin/interact/github.oauth.login/index`
+
+<figure>
+  <img src="/images/blog/authentication-plugins/github-login.png" class="has_border full_size"
+    alt="Figure 2: GoCD - Authorize Go Server on GitHub" id="mature_ci_cd_setup" title="GoCD - Authorize Go Server on GitHub" />
+  <figcaption>Figure 3: Authorize Go Server to access GitHub <span class="click_to_enlarge">(Click to enlarge)</span></figcaption>
+</figure>
+
+* Successful login
+
+<figure>
+  <img src="/images/blog/authentication-plugins/successful-login.png" class="has_border full_size"
+    alt="Figure 2: GoCD - On Successful Login" id="mature_ci_cd_setup" title="GoCD - On Successful Login" />
+  <figcaption>Figure 3: On successful login <span class="click_to_enlarge">(Click to enlarge)</span></figcaption>
+</figure>
+
+We hope plugin developers are able to use this feature to provide a better experience to their users.
```

Comment:
"We hope plugin developers are able to use this feature to provide a better experience to their users"

It will help if we are specific here. Like use end-point to build plugins to support their organizations authentication mechanism. 

Also mentioned what this end point supports (user search) and does not support